frame/election-provider-support/src/onchain.rs

```diff
@@ -130,11 +144,16 @@ fn elect_with<T: Config>(
 		DispatchClass::Mandatory,
 	);
 
-	Ok(to_supports(&staked))
+	let supports: OnChainBoundedSupportsOf<T> =
```

Comment:
As a follow-up, we should make this also be capable of truncating/sorting/error-ing. Then, on `do_elect` we don't need that extra `and_then`. If the `Fallback` is configured to error, it will error internally. Else, it can respect the bounds as cofnfigured.

It is reasonable to do this now, since it does not seem like much extra work to me.

Reply:
https://github.com/paritytech/substrate/pull/12436/commits/3a0f9efecb5e68d310d4330a04d59914bdb2ca5d

I did this here but removed it. The NPos solver returns an election result bounded by `desired_targets`. If we can guarantee that `desired_targets` is always less than `max_winners`, we can guarantee that this will never overflow. The only scenario where we will get this error is if Npos solver is not working as expected. Does this make sense? 
I thought I added a comment about it but apparently not. Will add.